pkg/aws/ec2/ec2.go

```diff
@@ -142,6 +159,61 @@ func (c *Client) describeNetworkInterfaces(ctx context.Context, subnets ipamType
 	return result, nil
 }
 
+// describeNetworkInterfacesFromInstances lists all ENIs matching filtered EC2 instances
+func (c *Client) describeNetworkInterfacesFromInstances(ctx context.Context) ([]ec2_types.NetworkInterface, error) {
+	ENIsFromInstances := make(map[string]*struct{})
```

Comment:
Nit: this doesn't need to be a pointer to a struct and can just be struct{}.
```suggestion
	ENIsFromInstances := make(map[string]struct{})
```

Reply:
ok changed.